eng/pipelines/templates/jobs/smoke-tests.yml

```diff
@@ -0,0 +1,167 @@
+        MacOs (AzureCloud):
+          OSVmImage: "macOS-10.15"
+          TestTargetFramework: netcoreapp2.1
+          SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+          ArmTemplateParameters: $(azureCloudArmParameters)
+        Windows_NetCoreApp (AzureUSGovernment):
+          OSVmImage: "windows-2019"
+          TestTargetFramework: netcoreapp2.1
+          SubscriptionConfiguration: $(sub-config-gov-test-resources)
+          ArmTemplateParameters: $(azureUSGovernmentArmParameters)
+        Windows_NetFramework (AzureUSGovernment):
+          OSVmImage: "windows-2019"
+          TestTargetFramework: net461
+          SubscriptionConfiguration: $(sub-config-gov-test-resources)
+          ArmTemplateParameters: $(azureUSGovernmentArmParameters)
+        Windows_NetCoreApp (AzureChinaCloud):
+          OSVmImage: "windows-2019"
+          TestTargetFramework: netcoreapp2.1
+          SubscriptionConfiguration: $(sub-config-cn-test-resources)
+          ArmTemplateParameters: $(azureChinaCloudArmParameters)
+        Windows_NetFramework (AzureChinaCloud):
+          OSVmImage: "windows-2019"
+          TestTargetFramework: net461
+          SubscriptionConfiguration: $(sub-config-cn-test-resources)
+          ArmTemplateParameters: $(azureChinaCloudArmParameters)
+
+    variables:
+      azureCloudArmParameters: "@{ storageEndpointSuffix = 'core.windows.net'; azureCloud = 'AzureCloud'; }"
+      azureUSGovernmentArmParameters: "@{ storageEndpointSuffix = 'core.usgovcloudapi.net'; azureCloud = 'AzureUSGovernment'; }"
+      azureChinaCloudArmParameters: "@{ storageEndpointSuffix = 'core.chinacloudapi.cn'; azureCloud = 'AzureChinaCloud'; }"
+
+    pool:
+      vmImage: $(OSVmImage)
+
+    steps:
+      - template: /eng/common/pipelines/templates/steps/verify-agent-os.yml
+
+      - task: PowerShell@2
+        ${{ if eq(parameters.Daily, true) }}:
+          displayName: Use latest dev feed package versions
+        ${{ if eq(parameters.Daily, false) }}:
+          displayName: Use latest package versions
+        inputs:
+          targetType: filePath
+          filePath: ./common/SmokeTests/SmokeTest/Update-Dependencies.ps1
+          workingDirectory: common/SmokeTests/SmokeTest
+          pwsh: true
+          ${{ if eq(parameters.Daily, true) }}:
+            arguments: -CI -Daily
+            displayName: Use latest daily package versions
+          ${{ if eq(parameters.Daily, false) }}:
+            arguments: -CI
+            displayName: Use latest GA/Beta package versions
+
+      - ${{ if eq(parameters.Daily, false) }}:
+        - download: current
+          artifact: ${{ parameters.ArtifactName }}
+          timeoutInMinutes: 5
+
+        - ${{ each artifact in parameters.Artifacts }}:
+          - ${{ if and(ne(variables['Skip.Release'], 'true'), ne(artifact.skipPublishPackage, 'true')) }}:
+            - pwsh: |
+                [xml]$csproj = Get-Content ./common/SmokeTests/SmokeTest/SmokeTest.csproj
```

Comment:
I was expecting to find this in the Update-Dependencies.ps1 script. I believe it would be a good idea to add an option to that script that take a directory with packages that will be used to override the versions of any matching packages.

Reply:
@weshaggard I would love to do it this way, but it's not feasible because we can't pass `parameters.Artifacts` down to a script, since azure pipelines templates can't do any yaml object conversion into script string fields. It's the same problem that has caused us to define arguments as powershell object string literals in the past. In this case, I didn't think that was a reasonable approach because it would have required changing every SDKs ci.yml file.

If you can think of a better solution, I'd be very happy :)

Reply:
Moved over. Was able to simplify the logic a lot. Wish I had thought of this yesterday :)